src/storage/src/hummock/sstable/multi_builder.rs

```diff
@@ -131,18 +154,35 @@ where
             let sstable_store = self.sstable_store.clone();
             let meta_clone = meta.clone();
             let policy = self.policy;
+            let uploading_size = self.uploading_size.clone();
+            let threshold = usize::MAX;
+            let uploading_sender = self.uploading_sender.clone();
             let upload_join_handle = tokio::spawn(async move {
-                if policy == CachePolicy::Fill {
-                    let sst = Sstable::new_with_data(table_id, meta_clone, data.clone())?;
-                    sstable_store.put(sst, data, CachePolicy::Fill).await
+                if let Some(sender) = uploading_sender && uploading_size.load(Ordering::Relaxed) > threshold {
+                    let (tx, rx) = oneshot::channel();
+                    sender.send(UploadRequest{
```

Comment:
Who is the receiver? How does the recceiver process UploadRequest?

Reply:
> Who is the receiver? How does the recceiver process UploadRequest?

receiver will do uploading step for sender

Reply:
> Who is the receiver? How does the recceiver process UploadRequest?

receiver is generated with sender

Reply:
Can you also include the receiver logic in this PR?

Reply:
> Can you also include the receiver logic in this PR?

ok

Reply:
> Can you also include the receiver logic in this PR?

just pushed a commit to implement receiver logic